otaclient/app/create_standby/rebuild_mode.py

```diff
@@ -61,95 +42,52 @@
-            # NOTE: check requests doc for details
-            self.proxies = {"http": proxy}
-
-    def _prepare_meta_files(self):
-        self.update_phase_tracker(wrapper.StatusProgressPhase.METADATA)
-        try:
-            for meta_f in self.metadata.get_img_metafiles():
-                meta_f_url = urljoin_ensure_base(self.url_base, quote(meta_f.file))
-                self._downloader.download(
-                    meta_f_url,
-                    self._recycle_folder / meta_f.file,
-                    digest=meta_f.hash,
-                    proxies=self.proxies,
-                    cookies=self.cookies,
-                    headers={
-                        OTAFileCacheControl.header_lower.value: OTAFileCacheControl.no_cache.value
-                    },
-                )
-        except HashVerificaitonError as e:
-            raise OTAMetaVerificationFailed from e
-        except DestinationNotAvailableError as e:
-            raise OTAErrorUnRecoverable from e
-        except DownloadError as e:
-            raise OTAMetaDownloadFailed from e
+        self._ota_tmp.mkdir(exist_ok=True)
+        # downloaded otameta files are stored under this folder
+        self._ota_tmp_image_meta_dir.mkdir(exist_ok=True)
 
     def _cal_and_prepare_delta(self):
-        self.update_phase_tracker(wrapper.StatusProgressPhase.REGULAR)
-
-        # TODO 2: old_reg is not used currently
         logger.info("generating delta...")
 
         regular_inf_fname = self.metadata.regular.file
         dir_inf_fname = self.metadata.directory.file
-        try:
-            delta_calculator = DeltaGenerator(
-                old_reg=Path(self.META_FOLDER) / regular_inf_fname,
-                new_reg=self._recycle_folder / regular_inf_fname,
-                new_dirs=self._recycle_folder / dir_inf_fname,
-                ref_root=self.reference_slot_mp,
-                recycle_folder=self._recycle_folder,
-                stats_collector=self.stats_collector,
-            )
-            self.delta_bundle = delta_calculator.get_delta()
-            self.stats_collector.set_total_regular_files(
-                self.delta_bundle.total_regular_num
-            )
-            logger.info(
-                f"total_regular_files_num={self.delta_bundle.total_regular_num}"
-            )
-        except Exception as e:
-            # TODO: define specific error for delta generator
-            raise UpdateDeltaGenerationFailed from e
+        delta_calculator = DeltaGenerator(
+            delta_src=self.active_slot_mp,
+            local_copy_dir=self._ota_tmp,
+            stats_collector=self.stats_collector,
+        )
+        delta_bundle = delta_calculator.calculate_and_process_delta(
+            delta_src_reg=Path(cfg.META_FOLDER) / regular_inf_fname,
+            new_reg=self._ota_tmp_image_meta_dir / regular_inf_fname,
+            new_dirs=self._ota_tmp_image_meta_dir / dir_inf_fname,
+        )
```

Comment:
Sorry, why is this modification necessary for this PR?

Reply:
**time-based retry strategy** depends on managing all the downloading tasks within the same **downloading tasks group**, to support **downloading tasks group**, delta needs to be pre-calculated before otaclient starts downloading. 
Delta generation is done by the create_standby module, so this API method is exposed by create_standby module.

Reply:
Please check `create_standby/interface.py` for more details.